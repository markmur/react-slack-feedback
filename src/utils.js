/**
 * Simple is object check.
 * @param item
 * @returns {boolean}
 */
function isObject(item) {
  return item && typeof item === 'object' && !Array.isArray(item)
}

/**
 * Deep merge two objects.
 * @param target
 * @param source
 */
export const merge = (target, source) => {
  if (isObject(target) && isObject(source)) {
    for (const key in source) {
      if (isObject(source[key])) {
        if (!target[key]) Object.assign(target, { [key]: {} })
        merge(target[key], source[key])
      } else {
        Object.assign(target, { [key]: source[key] })
      }
    }
  }
  return target
}

/**
 * Convert a hex value to an rgba value
 * @param {String} hex - 6 char hex string (#000000)
 * @param {Float} opacity - rgba opacity
 */
export const rgba = (hex, opacity = 1) => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)

  if (!result) return 'inherit'

  return `rgba(${parseInt(result[1], 16)}, ${parseInt(
    result[2],
    16
  )}, ${parseInt(result[3], 16)}, ${
    isNaN(parseInt(opacity, 10)) ? 1 : opacity
  })`
}

/**
 * Lookup an object property by dot notation
 * @param  {Object} obj - object to perform lookup
 * @param  {String} key - property location
 * @param  {Any} fallback - fallback if not found
 * @return {Any} returns value of lookup if found, otherwise undefined
 */
export const get = (obj, key, fallback) => {
  if (!obj && typeof obj !== 'object') return fallback
  return (
    key
      .split('.')
      .reduce((state, x) => (state && state[x] ? state[x] : null), obj) ||
    fallback
  )
}
