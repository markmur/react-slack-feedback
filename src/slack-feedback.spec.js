import React from 'react'
import { shallow } from 'enzyme'

import { default as SlackIcon } from './slack-icon'
import defaultTheme from './themes/default'
import SlackFeedback from './slack-feedback'
import translations from './translations'
import { merge } from './utils'

const __ = key => (key in translations ? translations[key] : key)

const DEFAULT_PROPS = {
  user: 'unit-tester',
  channel: '#tests',
  sentTimeout: 0
}

const render = props =>
  shallow(<SlackFeedback {...Object.assign({}, DEFAULT_PROPS, props)} />)

const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))

describe('SlackFeedback', () => {
  let component
  let find
  let trigger
  let submit
  let textarea
  let close
  let onSubmit
  let onImageUpload
  let onOpen
  let onClose
  let defaultPropCallbacks

  beforeEach(() => {
    onSubmit = jest.fn()
    onImageUpload = jest.fn()
    onOpen = jest.fn()
    onClose = jest.fn()

    defaultPropCallbacks = {
      onSubmit,
      onImageUpload,
      onOpen,
      onClose
    }

    component = render({
      ...defaultPropCallbacks
    })

    find = name => component.find(`styles__${name}`)

    trigger = find('Trigger')
    submit = find('SubmitButton')
    textarea = component.find('[name="message"]')
    close = component.find('.close')
  })

  it('should match the snapshots', () => {
    expect(component).toMatchSnapshot()
    expect(
      render({ disabled: true, ...defaultPropCallbacks })
    ).toMatchSnapshot()
    expect(
      render({ showChannel: false, ...defaultPropCallbacks })
    ).toMatchSnapshot()
    expect(
      render({ showIcon: false, ...defaultPropCallbacks })
    ).toMatchSnapshot()
  })

  describe('exports', () => {
    it('the default theme should be exported as defaultTheme', () => {
      expect(SlackFeedback.defaultTheme).toEqual(defaultTheme)
    })
    it('the slack icon should be exported as SlackIcon', () => {
      expect(SlackFeedback.SlackIcon).toEqual(SlackIcon)
    })
  })

  describe('translations', () => {
    it('should return the correct value from the translations', () => {
      expect(component.instance().translate('submit.text')).toEqual(
        'Send Feedback'
      )
    })

    it('should return an empty string if not found', () => {
      expect(component.instance().translate('not.available')).toEqual('')
    })

    it('should return the correct value from a translation prop', () => {
      component.setProps({ translations: { 'trigger.text': 'Feedback' } })
      expect(component.instance().translate('trigger.text')).toEqual('Feedback')
    })

    it('should return default translations if translation prop is provided', () => {
      component.setProps({ translations: { 'trigger.text': 'Feedback' } })
      expect(component.instance().translate('submit.text')).toEqual(
        'Send Feedback'
      )
    })
  })

  describe('presentational', () => {
    it('should show the channel by default', () => {
      expect(component.find('#channel')).toHaveLength(1)
    })

    it('should not show the channel if showChannel is false', () => {
      component.setProps({ showChannel: false })
      expect(component.find('#channel')).toHaveLength(0)

      // Reset the value
      component.setProps({ showChannel: true })
    })
    it('should display nothing if disabled', () => {
      component.setProps({ disabled: true })
      expect(component.html()).toBe(null)

      // Reset the value
      component.setProps({ disabled: false })
    })
  })

  describe('open/close', () => {
    it('should be closed by default', () => {
      expect(component.state('open')).toBe(false)
    })

    it('should fire close if toggle on state.open', () => {
      component.setState({ open: true })
      component.instance().toggle()
      expect(onClose).toHaveBeenCalled()
    })

    it('should open when the trigger is clicked and fire this.props.onOpen', () => {
      component.setState({ open: false })
      trigger.simulate('click')
      expect(component.state('open')).toBeTruthy()
      expect(onOpen).toHaveBeenCalled()
    })

    it('should set the open state to false on close click', () => {
      component.setState({ open: true })
      close.simulate('click')
      expect(component.state('open')).toEqual(false)
      expect(onClose).toHaveBeenCalled()
    })
  })

  describe('handleClickOutside', () => {
    it('should close if inside component', () => {
      component.setState({ open: true })
      close.simulate('click', {})
      expect(onClose).toHaveBeenCalled()
    })
  })

  describe('tabs', () => {
    const customTabs = [
      {
        value: 'unit',
        label: 'Unit'
      },
      {
        value: 'test',
        label: 'Test'
      }
    ]

    it('should select the first type by default', () => {
      expect(component.state('selectedType')).toEqual('bug')
    })

    it('should select the correct tab', () => {
      const newComponent = render({
        feedbackTypes: customTabs,
        defaultSelectedType: 'test',
        ...defaultPropCallbacks
      })
      expect(newComponent.state('selectedType')).toEqual('test')
    })
    it('should change the tab state on click', () => {
      const newComponent = render({
        feedbackTypes: customTabs,
        ...defaultPropCallbacks
      })

      expect(newComponent.state().selectedType).toBe(customTabs[0].value)

      newComponent
        .find('li')
        .at(1)
        .simulate('click')
      expect(newComponent.state().selectedType).toBe(customTabs[1].value)
    })
  })

  describe('onSubmit', () => {
    it('should set the sending state to true', () => {
      expect(component.state('sending')).toBe(false)
      submit.simulate('click')
      expect(component.state('sending')).toBe(true)
    })
    it('should set the default log level to "danger"', () => {
      submit.simulate('click')
      const payload = onSubmit.mock.calls[0][0]
      expect(payload.attachments[0].color).toEqual('danger')
    })

    it('should set the correct log level', () => {
      const nthPayload = n => onSubmit.mock.calls[n][0]

      component.setState({ selectedType: 'bug' })
      submit.simulate('click')
      expect(nthPayload(0).attachments[0].color).toEqual('danger')

      component.setState({ selectedType: 'feature' })
      submit.simulate('click')
      expect(nthPayload(1).attachments[0].color).toEqual('good')

      component.setState({ selectedType: 'improvement' })
      submit.simulate('click')
      expect(nthPayload(2).attachments[0].color).toEqual('warning')
    })

    it('should return warning level if selectedType invalid', () => {
      component.setState({ selectedType: 'invalid' })
      submit.simulate('click')
      const payload = onSubmit.mock.calls[0][0]
      expect(payload.attachments[0].color).toEqual('warning')
    })

    it('should return the correct payload', () => {
      const message = 'test'
      textarea.simulate('change', { target: { value: message } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')

      const expectedPayload = {
        attachments: [
          {
            fallback: `Feedback (${component.state('selectedType')})`,
            author_name: DEFAULT_PROPS.user,
            color: 'danger',
            title: component.state('selectedType'),
            title_link: document.location.href,
            text: message,
            footer: __('footer.text')
          }
        ]
      }

      expect(onSubmit).toHaveBeenCalledWith(
        expectedPayload,
        expect.any(Function),
        expect.any(Function)
      )
    })

    it('shuold attach image if available', () => {
      const message = 'test'
      textarea.simulate('change', { target: { value: message } })
      component.setState({ image: { url: 'fake-url' } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')

      const expectedPayload = {
        attachments: [
          {
            fallback: `Feedback (${component.state('selectedType')})`,
            author_name: DEFAULT_PROPS.user,
            color: 'danger',
            title: component.state('selectedType'),
            title_link: document.location.href,
            text: message,
            footer: __('footer.text'),
            image_url: 'fake-url'
          }
        ]
      }

      expect(onSubmit).toHaveBeenCalledWith(
        expectedPayload,
        expect.any(Function),
        expect.any(Function)
      )
    })

    it('should change sent state', () => {
      const message = 'test'
      onSubmit.mockImplementation((payload, success) => success())
      textarea.simulate('change', { target: { value: message } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')
      expect(component.state().sent).toBeTruthy()
      expect(component.state().sending).toBeFalsy()
      expect(component.state().error).toBeFalsy()
    })

    it('should reset sent state on success', async () => {
      const message = 'test'
      component.setProps({ sentTimeout: 0 })
      onSubmit.mockImplementation((payload, success) => success())
      textarea.simulate('change', { target: { value: message } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')
      expect(component.state().sent).toBeTruthy()
      await sleep(1)
      expect(component.state().sent).toBeFalsy()
    })

    it('should reset sent state on error', async () => {
      const message = 'test'
      component.setProps({ errorTimeout: 0 })
      onSubmit.mockImplementation((payload, success, error) =>
        error('test error')
      )
      textarea.simulate('change', { target: { value: message } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')
      expect(component.state().sending).toBeFalsy()
      expect(component.state().error).toBe('test error')
      await sleep(1)
      expect(component.state().error).toBeNull()
    })

    it('should change error state', () => {
      const message = 'test'
      onSubmit.mockImplementation((payload, success, error) =>
        error('TEST_ERROR')
      )
      textarea.simulate('change', { target: { value: message } })
      expect(component.state('message')).toBe(message)
      submit.simulate('click')
      expect(component.state().sent).toBeFalsy()
      expect(component.state().sending).toBeFalsy()
      expect(component.state().error).toBe('TEST_ERROR')
    })
  })

  describe('errors', () => {
    it('determineErrorType: should determine the correct error type', () => {
      const instance = component.instance()
      expect(instance.determineErrorType()).toEqual(__('error.unexpected'))
      expect(instance.determineErrorType('error')).toEqual('error')
      expect(instance.determineErrorType({ status: 400 })).toEqual(
        __('error.badrequest')
      )
      expect(instance.determineErrorType({ status: 403 })).toEqual(
        __('error.forbidden')
      )
      expect(instance.determineErrorType({ status: 404 })).toEqual(
        __('error.notfound')
      )
      expect(instance.determineErrorType({ status: 410 })).toEqual(
        __('error.archived')
      )
      expect(instance.determineErrorType({ status: 500 })).toEqual(
        __('error.internal')
      )
      expect(instance.determineErrorType({ status: 100000 })).toEqual(
        __('error.unexpected')
      )
    })
  })

  describe('theme', () => {
    const getTheme = () => component.find('ThemeProvider').props('theme').theme

    it('should have the default theme', () => {
      expect(getTheme()).toEqual(defaultTheme)
    })
    it('should deep merge a custom theme with the default', () => {
      const customTheme = {
        primary: 'blue',
        input: {
          color: 'red'
        }
      }

      component.setProps({
        theme: customTheme
      })

      const theme = getTheme()

      expect(theme.primary).toEqual(customTheme.primary)
      expect(theme.input.color).toEqual(customTheme.input.color)
      expect(theme).toEqual(merge({}, theme, customTheme))
    })
  })
})
