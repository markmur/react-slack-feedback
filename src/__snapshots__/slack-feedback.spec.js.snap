// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlackFeedback should match the snapshots 1`] = `
<ThemeProvider
  theme={
    Object {
      "button": Object {
        "boxShadow": "0 2px 8px 2px rgba(0, 135, 255, 0.2)",
      },
      "colors": Object {
        "background": "#f4f4f7",
        "border": "#d0d8e1",
        "error": "#ec3c3c",
        "primary": "#0088ff",
        "secondary": "#222c4f",
        "success": "#3dc86f",
        "text": "#333333",
      },
      "content": Object {
        "borderRadius": "4px",
        "boxShadow": "0 6px 30px 2px rgba(34, 44, 79, 0.3)",
        "fontSize": "14px",
        "padding": "0.6em",
      },
      "fontFamily": "-apple-system, BlinkMacSystemFont, Arial, Arial Unicode, \\"Helvetica Neue\\", Helvetica, \\"Hiragino Sans GB\\", \\"Microsoft YaHei\\", SimSun, sans-serif",
      "header": Object {
        "backgroundColor": "#222c4f",
        "color": "white",
      },
      "image": Object {
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "height": "140px",
      },
      "input": Object {
        "backgroundColor": "white",
        "borderRadius": "4px",
        "boxShadow": "0 0 8px rgba(0, 135, 255, 0.3)",
        "color": "#444444",
        "fontSize": "14px",
        "padding": "0.5em",
      },
      "label": Object {
        "color": "#0087ff",
        "fontSize": "11px",
      },
      "loader": Object {
        "color": "#ffffff",
        "size": "4em",
        "width": "3px",
      },
      "overlay": Object {
        "backgroundColor": "rgba(34, 44, 79, 0.4)",
      },
      "tab": Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "selectedBackgroundColor": "#0087ff",
        "selectedColor": "white",
      },
      "trigger": Object {
        "backgroundColor": "#ffffff",
        "border": "none",
        "borderRadius": "30px",
        "boxShadow": "0 3px 12px 1px rgba(34, 44, 79, 0.1)",
        "color": "#333333",
        "fontSize": "13px",
        "fontWeight": "normal",
        "hoverBackgroundColor": "inherit",
        "hoverBoxShadow": "0 6px 16px 2px rgba(0, 0, 0, 0.2)",
        "hoverColor": "inherit",
        "padding": "12px 1.25em",
      },
      "uploadButton": Object {
        "backgroundColor": "white",
        "border": "1px solid #d0d8e1",
        "hoverBackgroundColor": "rgba(255, 255, 255, 0.6)",
      },
    }
  }
>
  <styles__SlackFeedback
    className=""
  >
    <styles__Container
      className="fadeInUp"
    >
      <styles__Header>
        <icon />
         
        Send Feedback
        <styles__CloseButton
          className="close"
          onClick={[Function]}
        >
          close
        </styles__CloseButton>
      </styles__Header>
      <styles__Content>
        <span
          id="channel"
        >
          <styles__Label
            htmlFor="channel"
          >
            Channel
          </styles__Label>
          <styles__FormElement
            as="input"
            disabled={true}
            value="#tests"
          />
        </span>
        <styles__Label>
          Feedback Type
        </styles__Label>
        <styles__Tabs>
          <li
            className="selected"
            key="bug"
            onClick={[Function]}
            title="Bug"
          >
            Bug
          </li>
          <li
            className=""
            key="improvement"
            onClick={[Function]}
            title="Improvement"
          >
            Improvement
          </li>
          <li
            className=""
            key="feature"
            onClick={[Function]}
            title="Feature"
          >
            Feature
          </li>
        </styles__Tabs>
        <styles__Label>
          Message
        </styles__Label>
        <styles__FormElement
          as="textarea"
          name="message"
          onChange={[Function]}
          placeholder="Your message..."
          value=""
        />
        <styles__ImageUpload>
          <styles__UploadButton
            htmlFor="imageUpload"
          >
            Attach Image
          </styles__UploadButton>
          <styles__FormElement
            accept="image/*"
            as="input"
            id="imageUpload"
            onChange={[Function]}
            type="file"
          />
        </styles__ImageUpload>
        <styles__CheckboxContainer>
          <styles__Checkbox
            checked={true}
            id="sendURL"
            onChange={[Function]}
            type="checkbox"
          />
          <styles__CheckboxLabel
            htmlFor="sendURL"
          >
            Send url with feedback
          </styles__CheckboxLabel>
        </styles__CheckboxContainer>
        <styles__SubmitButton
          className=""
          disabled={true}
          onClick={[Function]}
        >
          Send Feedback
        </styles__SubmitButton>
      </styles__Content>
    </styles__Container>
    <styles__Trigger
      className=""
      onClick={[Function]}
    >
      <icon />
       
      Send Feedback
    </styles__Trigger>
  </styles__SlackFeedback>
</ThemeProvider>
`;

exports[`SlackFeedback should match the snapshots 2`] = `""`;

exports[`SlackFeedback should match the snapshots 3`] = `
<ThemeProvider
  theme={
    Object {
      "button": Object {
        "boxShadow": "0 2px 8px 2px rgba(0, 135, 255, 0.2)",
      },
      "colors": Object {
        "background": "#f4f4f7",
        "border": "#d0d8e1",
        "error": "#ec3c3c",
        "primary": "#0088ff",
        "secondary": "#222c4f",
        "success": "#3dc86f",
        "text": "#333333",
      },
      "content": Object {
        "borderRadius": "4px",
        "boxShadow": "0 6px 30px 2px rgba(34, 44, 79, 0.3)",
        "fontSize": "14px",
        "padding": "0.6em",
      },
      "fontFamily": "-apple-system, BlinkMacSystemFont, Arial, Arial Unicode, \\"Helvetica Neue\\", Helvetica, \\"Hiragino Sans GB\\", \\"Microsoft YaHei\\", SimSun, sans-serif",
      "header": Object {
        "backgroundColor": "#222c4f",
        "color": "white",
      },
      "image": Object {
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "height": "140px",
      },
      "input": Object {
        "backgroundColor": "white",
        "borderRadius": "4px",
        "boxShadow": "0 0 8px rgba(0, 135, 255, 0.3)",
        "color": "#444444",
        "fontSize": "14px",
        "padding": "0.5em",
      },
      "label": Object {
        "color": "#0087ff",
        "fontSize": "11px",
      },
      "loader": Object {
        "color": "#ffffff",
        "size": "4em",
        "width": "3px",
      },
      "overlay": Object {
        "backgroundColor": "rgba(34, 44, 79, 0.4)",
      },
      "tab": Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "selectedBackgroundColor": "#0087ff",
        "selectedColor": "white",
      },
      "trigger": Object {
        "backgroundColor": "#ffffff",
        "border": "none",
        "borderRadius": "30px",
        "boxShadow": "0 3px 12px 1px rgba(34, 44, 79, 0.1)",
        "color": "#333333",
        "fontSize": "13px",
        "fontWeight": "normal",
        "hoverBackgroundColor": "inherit",
        "hoverBoxShadow": "0 6px 16px 2px rgba(0, 0, 0, 0.2)",
        "hoverColor": "inherit",
        "padding": "12px 1.25em",
      },
      "uploadButton": Object {
        "backgroundColor": "white",
        "border": "1px solid #d0d8e1",
        "hoverBackgroundColor": "rgba(255, 255, 255, 0.6)",
      },
    }
  }
>
  <styles__SlackFeedback
    className=""
  >
    <styles__Container
      className="fadeInUp"
    >
      <styles__Header>
        <icon />
         
        Send Feedback
        <styles__CloseButton
          className="close"
          onClick={[Function]}
        >
          close
        </styles__CloseButton>
      </styles__Header>
      <styles__Content>
        <styles__Label>
          Feedback Type
        </styles__Label>
        <styles__Tabs>
          <li
            className="selected"
            key="bug"
            onClick={[Function]}
            title="Bug"
          >
            Bug
          </li>
          <li
            className=""
            key="improvement"
            onClick={[Function]}
            title="Improvement"
          >
            Improvement
          </li>
          <li
            className=""
            key="feature"
            onClick={[Function]}
            title="Feature"
          >
            Feature
          </li>
        </styles__Tabs>
        <styles__Label>
          Message
        </styles__Label>
        <styles__FormElement
          as="textarea"
          name="message"
          onChange={[Function]}
          placeholder="Your message..."
          value=""
        />
        <styles__ImageUpload>
          <styles__UploadButton
            htmlFor="imageUpload"
          >
            Attach Image
          </styles__UploadButton>
          <styles__FormElement
            accept="image/*"
            as="input"
            id="imageUpload"
            onChange={[Function]}
            type="file"
          />
        </styles__ImageUpload>
        <styles__CheckboxContainer>
          <styles__Checkbox
            checked={true}
            id="sendURL"
            onChange={[Function]}
            type="checkbox"
          />
          <styles__CheckboxLabel
            htmlFor="sendURL"
          >
            Send url with feedback
          </styles__CheckboxLabel>
        </styles__CheckboxContainer>
        <styles__SubmitButton
          className=""
          disabled={true}
          onClick={[Function]}
        >
          Send Feedback
        </styles__SubmitButton>
      </styles__Content>
    </styles__Container>
    <styles__Trigger
      className=""
      onClick={[Function]}
    >
      <icon />
       
      Send Feedback
    </styles__Trigger>
  </styles__SlackFeedback>
</ThemeProvider>
`;

exports[`SlackFeedback should match the snapshots 4`] = `
<ThemeProvider
  theme={
    Object {
      "button": Object {
        "boxShadow": "0 2px 8px 2px rgba(0, 135, 255, 0.2)",
      },
      "colors": Object {
        "background": "#f4f4f7",
        "border": "#d0d8e1",
        "error": "#ec3c3c",
        "primary": "#0088ff",
        "secondary": "#222c4f",
        "success": "#3dc86f",
        "text": "#333333",
      },
      "content": Object {
        "borderRadius": "4px",
        "boxShadow": "0 6px 30px 2px rgba(34, 44, 79, 0.3)",
        "fontSize": "14px",
        "padding": "0.6em",
      },
      "fontFamily": "-apple-system, BlinkMacSystemFont, Arial, Arial Unicode, \\"Helvetica Neue\\", Helvetica, \\"Hiragino Sans GB\\", \\"Microsoft YaHei\\", SimSun, sans-serif",
      "header": Object {
        "backgroundColor": "#222c4f",
        "color": "white",
      },
      "image": Object {
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "height": "140px",
      },
      "input": Object {
        "backgroundColor": "white",
        "borderRadius": "4px",
        "boxShadow": "0 0 8px rgba(0, 135, 255, 0.3)",
        "color": "#444444",
        "fontSize": "14px",
        "padding": "0.5em",
      },
      "label": Object {
        "color": "#0087ff",
        "fontSize": "11px",
      },
      "loader": Object {
        "color": "#ffffff",
        "size": "4em",
        "width": "3px",
      },
      "overlay": Object {
        "backgroundColor": "rgba(34, 44, 79, 0.4)",
      },
      "tab": Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "border": "1px solid #d0d8e1",
        "borderRadius": "4px",
        "selectedBackgroundColor": "#0087ff",
        "selectedColor": "white",
      },
      "trigger": Object {
        "backgroundColor": "#ffffff",
        "border": "none",
        "borderRadius": "30px",
        "boxShadow": "0 3px 12px 1px rgba(34, 44, 79, 0.1)",
        "color": "#333333",
        "fontSize": "13px",
        "fontWeight": "normal",
        "hoverBackgroundColor": "inherit",
        "hoverBoxShadow": "0 6px 16px 2px rgba(0, 0, 0, 0.2)",
        "hoverColor": "inherit",
        "padding": "12px 1.25em",
      },
      "uploadButton": Object {
        "backgroundColor": "white",
        "border": "1px solid #d0d8e1",
        "hoverBackgroundColor": "rgba(255, 255, 255, 0.6)",
      },
    }
  }
>
  <styles__SlackFeedback
    className=""
  >
    <styles__Container
      className="fadeInUp"
    >
      <styles__Header>
         
        Send Feedback
        <styles__CloseButton
          className="close"
          onClick={[Function]}
        >
          close
        </styles__CloseButton>
      </styles__Header>
      <styles__Content>
        <span
          id="channel"
        >
          <styles__Label
            htmlFor="channel"
          >
            Channel
          </styles__Label>
          <styles__FormElement
            as="input"
            disabled={true}
            value="#tests"
          />
        </span>
        <styles__Label>
          Feedback Type
        </styles__Label>
        <styles__Tabs>
          <li
            className="selected"
            key="bug"
            onClick={[Function]}
            title="Bug"
          >
            Bug
          </li>
          <li
            className=""
            key="improvement"
            onClick={[Function]}
            title="Improvement"
          >
            Improvement
          </li>
          <li
            className=""
            key="feature"
            onClick={[Function]}
            title="Feature"
          >
            Feature
          </li>
        </styles__Tabs>
        <styles__Label>
          Message
        </styles__Label>
        <styles__FormElement
          as="textarea"
          name="message"
          onChange={[Function]}
          placeholder="Your message..."
          value=""
        />
        <styles__ImageUpload>
          <styles__UploadButton
            htmlFor="imageUpload"
          >
            Attach Image
          </styles__UploadButton>
          <styles__FormElement
            accept="image/*"
            as="input"
            id="imageUpload"
            onChange={[Function]}
            type="file"
          />
        </styles__ImageUpload>
        <styles__CheckboxContainer>
          <styles__Checkbox
            checked={true}
            id="sendURL"
            onChange={[Function]}
            type="checkbox"
          />
          <styles__CheckboxLabel
            htmlFor="sendURL"
          >
            Send url with feedback
          </styles__CheckboxLabel>
        </styles__CheckboxContainer>
        <styles__SubmitButton
          className=""
          disabled={true}
          onClick={[Function]}
        >
          Send Feedback
        </styles__SubmitButton>
      </styles__Content>
    </styles__Container>
    <styles__Trigger
      className=""
      onClick={[Function]}
    >
       
      Send Feedback
    </styles__Trigger>
  </styles__SlackFeedback>
</ThemeProvider>
`;
